lib/pipeline.js

```diff
@@ -1145,6 +1147,66 @@ class PipelineModel extends BaseModel {
             .then(() => removeChildPipelines()) // remove pr events
             .then(() => super.remove()); // remove pipeline
     }
+
+    /**
+     * getReports for this pipeline
+     * @method getReports
+     * @param  {Object}   [config]              Configuration object
+     * @param  {String}   [config.startTime]    Look at events created after this startTime
+     * @param  {String}   [config.endTime]      Look at events created before this endTime
+     * @return {Promise}  Resolves to array of metrics for events belong to this pipeline
+     */
+    async getMetrics(config) {
+        // if not config pass in then get all
+        const startTime = config ? config.startTime : null;
+        const endTime = config ? config.endTime : null;
+
+        // Get events during this time range
+        const events = await this.getEvents({
+            startTime,
+            endTime
+        });
+
+        // Calculate event duration by using max endTime - min startTime of builds
+        // that belong this event during this time range
+        const eventDuration = async (event) => {
+            const builds = await event.getBuilds({
+                startTime,
+                endTime
+            });
+
+            // Somehow this event doesn't have any builds
+            if (!builds) {
+                return 0;
+            }
+
+            let minStartTime = new Date(builds[0].startTime);
```

Comment:
Can make this block of code cleaner using `map` and `sort`, but will not be as efficient.